app/file.go

```diff
@@ -1104,11 +1118,33 @@ func (a *App) generatePreviewImage(img image.Image, previewPath string, width in
 }
 
 func (a *App) GetFileInfo(fileId string) (*model.FileInfo, *model.AppError) {
-	return a.Srv().Store.FileInfo().Get(fileId)
+	fileInfo, err := a.Srv().Store.FileInfo().Get(fileId)
+	if err != nil {
+		var nfErr *store.ErrNotFound
+		switch {
+		case errors.As(err, &nfErr):
+			return nil, model.NewAppError("GetFileInfo", "app.file_info.get.app_error", nil, nfErr.Error(), http.StatusNotFound)
+		default:
+			return nil, model.NewAppError("GetFileInfo", "app.file_info.get.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+	}
+
+	return fileInfo, nil
 }
 
 func (a *App) GetFileInfos(page, perPage int, opt *model.GetFileInfosOptions) ([]*model.FileInfo, *model.AppError) {
-	return a.Srv().Store.FileInfo().GetWithOptions(page, perPage, opt)
+	fileInfos, err := a.Srv().Store.FileInfo().GetWithOptions(page, perPage, opt)
+	if err != nil {
+		var ltErr *store.ErrLimitExceeded
```

Comment:
Should we handling here the `NewErrInvalidInput` error and return a bad request instead of an internal server error?

Reply:
yep Jesus, we just changed to `ErrLimitExceeded` but in the app layer we are returning the bad request status, so it is ok